packages/framework/src/Command/FixExtendedClassesAnnotationsCommand.php

```diff
@@ -113,11 +120,118 @@ protected function getAnnotationsReplacementsMap(array $classExtensionMap): arra
     {
         $annotationsReplacementsMap = [];
         foreach ($classExtensionMap as $frameworkClass => $projectClass) {
-            $index = '/\\\\' . str_replace('\\', '\\\\', $frameworkClass) . '\b/';
+            $index = $this->getEscapedFqcnWithLeadingSlashPattern($frameworkClass);
             $value = '\\' . $projectClass;
             $annotationsReplacementsMap[$index] = $value;
         }
 
         return $annotationsReplacementsMap;
     }
+
+    protected function addPropertyAnnotationsToProjectClasses()
+    {
+        $classExtensionMap = $this->classExtensionRegistry->getClassExtensionMap();
+        foreach ($classExtensionMap as $frameworkClass => $projectClass) {
+            $frameworkClassBetterReflection = $this->getBetterReflectionClass($frameworkClass);
+            foreach ($frameworkClassBetterReflection->getProperties() as $property) {
+                $this->addPropertyAnnotationToProjectClassIfNecessary($property, $projectClass, array_keys($classExtensionMap));
+            }
+        }
+    }
+
+    /**
+     * @param \Roave\BetterReflection\Reflection\ReflectionProperty $reflectionProperty
+     * @param string $projectClass
+     * @param string[] $extendedFrameworkClasses
+     */
+    protected function addPropertyAnnotationToProjectClassIfNecessary(
+        ReflectionProperty $reflectionProperty,
+        string $projectClass,
+        array $extendedFrameworkClasses
+    ) {
+        foreach ($extendedFrameworkClasses as $extendedFrameworkClass) {
+            $isPropertyOfTypeThatIsExtendedInProject = preg_match(
+                $this->getEscapedFqcnWithLeadingSlashPattern($extendedFrameworkClass),
+                $reflectionProperty->getDocComment()
+            );
+            if ($isPropertyOfTypeThatIsExtendedInProject) {
+                $this->addPropertyAnnotationToClass($reflectionProperty, $projectClass);
+            }
+        }
+    }
+
+    /**
+     * @param string $fullyQualifiedClassName
+     * @return string
+     */
+    protected function getEscapedFqcnWithLeadingSlashPattern(string $fullyQualifiedClassName): string
+    {
+        return '/\\\\' . str_replace('\\', '\\\\', $fullyQualifiedClassName) . '\b/';
```

Comment:
can you please comment the "\b" in the regex? it's not clear at the first sight

Reply:
added in 551ddd64a, I hope it is understandable :smile:

Reply:
Perfectly clear. Just small typos